File: JUnitTestingLastAct/src/Singleton/SingletonTestRunner.java
package Singleton;

import org.testng.annotations.Test;

import java.util.logging.Logger;

import static org.testng.AssertJUnit.*;

public class SingletonTestRunner {

    private final Logger logger = Logger.getLogger(SingletonTestRunner.class.getName());

    @Test
    public final void stringIsSame() {
        assertTrue(SingletonTestCase.booleanMessage("This is a Singleton"));
        this.logger.info("Singleton has the same String");
    }

    @Test
    public final void stringIsNotSame() {
        assertFalse(SingletonTestCase.booleanMessage("This is a Singleton"));
        this.logger.info("Singleton does not produce the same String");
    }

    @Test
    public final void instanceExists() {
        assertNotNull(SingletonTestCase.objectExists());
        this.logger.info("The object exists.");
    }

    @Test
    public final void sameObject() {
        assertTrue(SingletonTestCase.isSame());
        this.logger.info("Same Object");
    }

    @Test
    public final void compareInstance() {
        Singleton singleton1 = Singleton.Instance;
        Singleton singleton2 = Singleton.Instance;
        assertTrue(SingletonTestCase.allCreatedInstanceIsSame(singleton1, singleton2));
        this.logger.info("Same Object");
    }
}

File: KotlinTesting/src/Test/Creational/SingletonTestRunner.kt
package Singleton

import Singleton.SingletonTestCase.allCreatedInstanceIsSame
import Singleton.SingletonTestCase.booleanMessage
import Singleton.SingletonTestCase.isSame
import Singleton.SingletonTestCase.objectExists
import org.junit.Test
import java.util.logging.Logger
import kotlin.test.assertFalse
import kotlin.test.assertNotNull
import kotlin.test.assertTrue

class SingletonTestRunner {
    private val logger = Logger.getLogger(SingletonTestRunner::class.java.name)

    @Test
    fun instanceExists() {
        assertNotNull(objectExists())
        logger.info("The object exists.")
    }

    @Test
    fun stringResultIsSame() {
        assertTrue(booleanMessage("This is a Singleton"))
        logger.info("Singleton has the same String")
    }

    @Test
    fun stringResultIsNotSame() {
        assertFalse(booleanMessage("This is a Singleton"))
        logger.info("Singleton does not produce the same String")
    }

    @Test
    fun sameObject() {
        assertTrue(isSame)
        logger.info("Same Object")
    }

    @Test
    fun compareInstance() {
        val singleton1 = Singleton.Instance
        val singleton2 = Singleton.Instance
        assertTrue(allCreatedInstanceIsSame(singleton1!!, singleton2!!))
        logger.info("Same Object")
    }
}